### ISSUE TITLE
Add: `Link` Component

### ISSUE THREAD

**Issue by fabiankaegy** [10up/block-components#62]
**Is your enhancement related to a problem? Please describe.**

A common pattern that we need to solve for in our blocks is the ability to have one or multiple CTA/Link areas inside of a Custom Block. These CTAs/Links may be optional but when an editor chooses to use the Link they need to provide both the Link Text **and** also the URL. 

**Describe the solution you'd like**

Create a new Component called `Link` that allows developers to pass a `text`, `url`, `onTextChange`, `onUrlSelect` and other props that may be passed through to the underlying `RichText` component like a `className etc.`

```js
const { linkText, linkUrl } = attributes;

<Link 
    text={linkText}
    onTextChange={ value => setAttributes( { linkText: value } ) }
    placeholder={ "Insert Link here..." }
    url={linkUrl}
    onLinkSelect={ value => setAttributes( { linkUrl: value } ) }
    className={`custom-class-name`}
/>
```

The Component should be aware of whether the current block is selected and not show when no text and no URL are provided and the block is not currently selected. When either of the values is shown but the other one isn't there should be an inline warning indication letting the user know that they are not done entering data into the component. 

When a user interacts with the link in the editor the `LinkControl` component should open up and allow editors to choose what they want to link to. 

**Designs**

/
 
**Describe alternatives you've considered**

/

**Additional context**

/

**Comment by fabiankaegy** [comment]
The PR #48 is an initial version of this component that we have previously used for this. The idea is to take the code from that and use it as the foundation for the new `Link` Component.

**Comment by fabiankaegy** [comment]
Another thing to consider here is how we can incorporate screenreader text to the links

**Comment by fabiankaegy** [comment]
I'd love to also figure out how we can make use of more of the options of the `LinkControl` in this component. 

`LinkControl` on its own allows you to define whether you want to open the link in a new tab and other boolean values. See the Docs here: https://github.com/WordPress/gutenberg/blob/55513110f87524777c54668407f897945f3905d2/packages/block-editor/src/components/link-control/README.md#settings
